### DISCUSSION TITLE
`set-output` command is deprecated

### DISCUSSION THREAD

**Discussion by liborty** [community/community#36930] [Actions] [answered]
My tests produce some `stdout`(put) .
I started getting the above warnings.
After searching hard for the solution (why not put it right in the warning message?),
I appended `>> $GITHUB_OUTPUT` to redirect my `stdout` to it.
However, instead of warnings, I now get outright errors that the output is failing.
Have I misunderstood something?
What is the remedy?

**Comment by Patstar210** [comment]
Not sure but it's coming from someone logging in into account through my platform

**Comment by antonblr** [comment, marked as answer]
Hey @liborty - the best thing to have when asking for help is to give specific examples. Could you paster your workflow yaml here? 

It could be that `set-output` warning is indirect and comes from the actions used in your workflow. i.e. [actions-rs/toolchain](https://github.com/actions-rs/toolchain/issues/221) haven't addressed that warning yet.

**Reply by liborty** [reply]
I have deleted the redirect now, as it causes everything to fail.
I suspect it may have something to do with `actions-rs/toolchain@v1` (rust actions) being badly out of date.
The `cargo` command (rust equivalent of make) obviously produces some output (the executables or tests output).
But I do not know for sure if it is `cargo` that is causing it. I am reluctant to be changing my standard scripts any further, 
just because github actions demand some weird redirections. This was the failed attempt:

`cargo test --release -- --nocapture --color always --test-threads=1  >> $GITHUB_OUTPUT`

also, I want  the output of the tests to be visible in the actions log file. That is why I run them.

**Reply by airtower-luna** [reply]
As @antonblr wrote, it's hard to give advice without seeing your actual workflow. Please quote at least the relevant step(s), not just a command.

What I can say about the redirect is that I'm pretty sure `cargo` doesn't produce the `name=value` format you need to set outputs. Most likely you'll need to use the same [format with a delimiter](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings) as for multi-line environment variables.

**Reply by liborty** [reply]
I get these warnings on all my actions that used to be just fine. For example, the following gives me this warning four times:
`The  set-output command is deprecated and will be disabled soon. ...`:

```
name: test

on: push

jobs:
  build:  
    name: tokencrypt 
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: actions-rs/toolchain@v1    
      with:
        toolchain: stable
    - name: install
      run: source ./install 
    - name: tests
      run: | 
        crptest testing > test.log
        echo >> test.log
        packtest testing >> test.log            
    - name: commitlog
    ....... credentials follow......

**Reply by airtower-luna** [reply]
Where's the `cargo` command you mentioned?

Also third-party action might cause those warnings, too, if they do you'll have to update or (if there is no newer version) ask the developer(s) to fix the problem.

**Reply by antonblr** [reply]
@liborty - From what I see only `actions-rs/toolchain` will throw those warnings (there is an open issue with them, see my first comment).

**Reply by liborty** [reply]
The cargo commands here are within the ./install script.
Anyway, thank you for the answer.
I will keep the fingers crossed then that the developers manage to update `actions-rs/toolchain@v1` before the 'soon' change makes all the rust actions fail.
